File: app/src/main/java/uk/lobsterdoodle/namepicker/addgroup/NameCard.java
package uk.lobsterdoodle.namepicker.addgroup;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import uk.lobsterdoodle.namepicker.R;
import uk.lobsterdoodle.namepicker.application.util.As;
import uk.lobsterdoodle.namepicker.edit.NameCardActions;
import uk.lobsterdoodle.namepicker.model.Name;

public class NameCard extends CardView {

    @BindView(R.id.name_card_name)
    TextView name;

    @BindView(R.id.name_card_delete_icon)
    ImageView delete;

    public NameCard(Context context) {
        super(context);
        init();
    }

    public NameCard(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public NameCard(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        ButterKnife.bind(this, LayoutInflater.from(getContext()).inflate(R.layout.name_card, this, true));
        setCardBackgroundColor(getContext().getResources().getColor(android.R.color.white));
        setUseCompatPadding(true);
        setCardElevation(As.px(getContext(), 2));
    }

    public void bind(NameCardActions nameCardActions, Name name) {
        this.name.setText(name.getName());
        delete.setOnClickListener(v -> nameCardActions.deleteName(name.getId()));
    }
}

File: app/src/main/kotlin/uk/lobsterdoodle/namepicker/addgroup/NameCard.kt
package uk.lobsterdoodle.namepicker.addgroup

import android.content.Context
import androidx.cardview.widget.CardView
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.TextView
import uk.lobsterdoodle.namepicker.R
import uk.lobsterdoodle.namepicker.application.util.As
import uk.lobsterdoodle.namepicker.databinding.NameCardBinding
import uk.lobsterdoodle.namepicker.edit.NameCardActions
import uk.lobsterdoodle.namepicker.model.Name

class NameCard : CardView {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    private val binding: NameCardBinding =
        NameCardBinding.inflate(LayoutInflater.from(context), this, true)

    init {
        setCardBackgroundColor(context.resources.getColor(android.R.color.white))
        useCompatPadding = true
        cardElevation = As.px(context, 2).toFloat()
    }

    fun bind(nameCardActions: NameCardActions, name: Name) {
        binding.nameCardName.text = name.name
        binding.nameCardDeleteIcon.setOnClickListener { nameCardActions.deleteName(name.id) }
    }
}
